virtio-devices/src/vhost_user/vu_common_ctrl.rs

```diff
@@ -143,6 +143,15 @@ impl VhostUserHandle {
 
         // Setup for inflight I/O tracking shared memory.
         if let Some(inflight) = inflight {
+            // Send set_vring_base here, since it could tell backends, like SPDK,
+            // how many virt queues to be handled, which backend required to know
+            // at early stage.
+            for i in 0..queues.len() {
+                self.vu
+                    .set_vring_base(i, 0)
+                    .map_err(Error::VhostUserSetVringBase)?;
+            }
+
```

Comment:
So I like the idea but let's go one step further, and make sure we always set the vring base first.
That means before going through the inflight code (line 143), I would loop over the queues and call into:
```rust
for (queue_index, queue) in queues.into_iter().enumerate() {
    self.vu
        .set_vring_base(
            queue_index,
            queue
                .used_index_from_memory(mem)
                .map_err(Error::GetAvailableIndex)?,
        )
    .map_err(Error::VhostUserSetVringBase)?;
}
```

And I would remove lines 208 through 215, since this would be redundant.